dask/array/tests/test_creation.py

```diff
@@ -119,9 +120,36 @@ def test_arange():
     nparr = np.arange(0, -1, 0.5)
     assert_eq(darr, nparr)
 
-
-def test_arange_has_dtype():
-    assert da.arange(5, chunks=2).dtype == np.arange(5).dtype
+    # Unexpected or missing kwargs
+    with assert_raises(TypeError):
+        da.arange(10, chunks=-1, unexpected=True)
+    with assert_raises(TypeError):
+        da.arange(10)
+
+
+@pytest.mark.parametrize("start,stop,step,dtype", [
+    (0, 1, 1, None),  # int64
+    (1.5, 2, 1, None),  # float64
+    (1, 2.5, 1, None),  # float64
+    (1, 2, .5, None),  # float64
+    (np.float32(1), np.float32(2), np.float32(1), None),  # promoted to float64
+    (np.int32(1), np.int32(2), np.int32(1), None),  # promoted to int64
+    (np.uint32(1), np.uint32(2), np.uint32(1), None),  # promoted to int64
+    (np.uint64(1), np.uint64(2), np.uint64(1), None),  # promoted to float64
+    (np.uint32(1), np.uint32(2), np.uint32(1), np.uint32),
+    (np.uint64(1), np.uint64(2), np.uint64(1), np.uint64),
+    # numpy.arange gives unexpected results
+    # https://github.com/numpy/numpy/issues/11505
+    # (1j, 2, 1, None),
+    # (1, 2j, 1, None),
+    # (1, 2, 1j, None),
+    # (1+2j, 2+3j, 1+.1j, None),
+])
+def test_arange_dtypes(start, stop, step, dtype):
+    a_np = np.arange(start, stop, step, dtype=dtype)
+    a_da = da.arange(start, stop, step, dtype=dtype, chunks=-1)
+    assert a_np.dtype == a_da.dtype
```

Comment:
I think that this should be checked automatically in `assert_eq`.  If not then we should put it there instead.

Reply:
I know it is checked, however I don't know (and I don't want to rely upon) if it is checked before or after compute() which in this case makes a lot of difference

Reply:
I believe that it is checked both before and after.  If I'm wrong and it isn't then it should be.